commit f9ed41cb2487f0e5e8df2cf52f60fd4dc685d534 (refs/changes/48/548/3)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-09-29T17:50:22-07:00 (3 years ago)
    
    joyent/node-cueball#27 dynamic resolver mode broken
    joyent/node-cueball#28 cbresolve broken in 0.5.0

diff --git a/.npmignore b/.npmignore
index f54a5da..bf7527d 100644
--- a/.npmignore
+++ b/.npmignore
@@ -7,3 +7,4 @@ test
 tools
 coverage
 demo-*.js
+simulator*.js
diff --git a/bin/cbresolve b/bin/cbresolve
index 21d6d18..8f355be 100755
--- a/bin/cbresolve
+++ b/bin/cbresolve
@@ -208,8 +208,13 @@ function main()
 	cbrResolver.on('removed', cbrPrintRemoved);
 
 	if (!cbrFollow) {
-		cbrResolver.onState('running', cbrRunning);
-		cbrResolver.onState('failed', cbrFailed);
+		cbrResolver.on('stateChanged', function (st) {
+			if (st === 'running') {
+				cbrRunning();
+			} else if (st === 'failed') {
+				cbrFailed();
+			}
+		});
 		process.on('exit', function (code) {
 			if (code === 0) {
 				mod_assert.ok(cbrDone, 'premature exit');
diff --git a/lib/resolver.js b/lib/resolver.js
index 0b7bb66..3ce30ac 100644
--- a/lib/resolver.js
+++ b/lib/resolver.js
@@ -21,7 +21,7 @@ module.exports = {
 	parseIpOrDomain: parseIpOrDomain
 };
 
-const mod_nsc = require('named-client');
+const mod_nsc = require('mname-client');
 const mod_events = require('events');
 const mod_net = require('net');
 const mod_util = require('util');
@@ -304,13 +304,14 @@ CueBallDNSResolver.prototype.state_check_ns = function (S) {
 		this.r_bootstrap =
 		    CueBallDNSResolver.bootstrapResolvers[notIp[0]];
 		if (this.r_bootstrap === undefined) {
-			this.r_bootstrap = new CueBallDNSResolver({
+			var res = new CueBallDNSResolver({
 				domain: notIp[0],
 				service: '_dns._udp',
 				defaultPort: 53,
 				log: this.r_log,
 				recovery: this.r_recovery
 			});
+			this.r_bootstrap = res.r_fsm;
 			CueBallDNSResolver.bootstrapResolvers[notIp[0]] =
 			    this.r_bootstrap;
 		}
diff --git a/package.json b/package.json
index 9448f88..bb5d277 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "cueball",
-  "version": "0.5.0",
+  "version": "1.0.0",
   "description": "",
   "main": "lib/index.js",
   "dependencies": {
@@ -10,7 +10,7 @@
     "extsprintf": ">=1.3.0 <2.0.0",
     "ipaddr.js": ">=1.1.0 <2.0.0",
     "mooremachine": ">=2.0.0 <3.0.0",
-    "named-client": "git+https://github.com/arekinath/node-named-client.git#v0.3.6",
+    "mname-client": ">=0.3.6 <0.4.0",
     "node-uuid": ">=1.4.7 <2.0.0",
     "posix-getopt": ">=1.2.0 <2.0.0",
     "restify-clients": ">=1.1.2 <2.0.0",
